### DISCUSSION TITLE
Deploy a Netdata container for offline system

### DISCUSSION THREAD

**Discussion by kavishgr** [netdata/netdata#15456] [Q&A]
I'm trying the exact same thing as mentioned in this [section](https://learn.netdata.cloud/docs/installing/offline-systems) but inside a container instead. I came up with the following **Dockerfile**:

```
FROM alpine:latest

# Set the working directory inside the container
WORKDIR /netdata-on-prem

# Copy the install.sh and netdata-x86.gz.run files into the container's working directory
COPY sha256sums.txt channel install.sh kickstart.sh netdata-x86_64-latest.gz.run /netdata-on-prem/

# Make install.sh executable (assuming it needs to be executed)
RUN chmod +x install.sh

# Run the install.sh script which will install and configure Netdata
RUN ./install.sh --no-updates  --disable-telemetry

# Expose the ports that Netdata uses (19999 by default for web UI and monitoring)
EXPOSE 19999

# Set the command to start Netdata when the container runs
CMD ["/opt/netdata/bin/netdata", "-D"]
```

Build the image:

```
docker build -t netdata-onprem .
```

Running the container(got the command in the docker install section from the [documentation](https://learn.netdata.cloud/docs/installing/docker)):

```
docker run -d --name=netdata \
  --pid=host \
  --network=host \
  -v netdataconfig:/etc/netdata \
  -v netdatalib:/var/lib/netdata \
  -v netdatacache:/var/cache/netdata \
  -v /etc/passwd:/host/etc/passwd:ro \
  -v /etc/group:/host/etc/group:ro \
  -v /proc:/host/proc:ro \
  -v /sys:/host/sys:ro \
  -v /etc/os-release:/host/etc/os-release:ro \
  -v /var/run/docker.sock:/var/run/docker.sock:ro \
  --restart unless-stopped \
  --cap-add SYS_PTRACE \
  --cap-add SYS_ADMIN \
  --security-opt apparmor=unconfined \
  netdata-onprem
  ```

Currently, I have the Netdata agent up and running, and there's an accessible dashboard at `http://localhost:19999/`. However, I'm unsure if I've built and run the image/container correctly. Unfortunately, I'm not receiving any metrics for Docker containers. Do I need a plugin/chart to enable this functionality? 

The docs are very clear and state that all collectors are automatically installed with every Netdata installation. There's no need to manually install collectors to collect metrics from new sources. .

Here's a screenshot:

<img width="1278" alt="Screenshot 2023-07-20" src="https://github.com/netdata/netdata/assets/45298829/a7f6bac1-d9bd-415b-b3a0-05c27b529daf">

**Comment by ilyam8** [comment]
Hey, @kavishgr. You are correct, you haven't built the image correctly. You need to use ENTRYPOINT and not CMD. The entrypoint should be [run.sh](https://github.com/netdata/netdata/blob/master/packaging/docker/run.sh).

**Comment by ilyam8** [comment]
@kavishgr what is the reason you don't use our official docker image? What modification do you need to do?
